hmrc-mongo-play-28/src/main/scala/uk/gov/hmrc/mongo/play/json/PlayMongoRepository.scala

```diff
@@ -45,21 +45,36 @@ class PlayMongoRepository[A: ClassTag](
   final val indexes: Seq[IndexModel],
   final val optSchema: Option[BsonDocument] = None,
   replaceIndexes: Boolean = false,
-  extraCodecs: Seq[Codec[_]] = Seq.empty
+  extraCodecs   : Seq[Codec[_]] = Seq.empty
 )(implicit ec: ExecutionContext)
     extends MongoDatabaseCollection {
 
+  private val logger = play.api.Logger(getClass)
+
   lazy val collection: MongoCollection[A] =
     CollectionFactory.collection(mongoComponent.database, collectionName, domainFormat, extraCodecs)
 
-  Await.result(ensureIndexes, 5.seconds)
+  /** Can be overridden if the repository manages it's own data cleanup */
+  protected[mongo] lazy val requiresTtlIndex = true
```

Comment:
I'm not sure what benefits setting this to true provides. If `checkType = false` then this will just return a map with the values set to `TtlState.TypeCheckSkipped`. The indexes don't seem to be logged?

Reply:
`checkType=false` skips the expensive check. The presence of ttl indexes is still checked for. Note, that tests use `checkType=true`